pkg/apis/policy/v1alpha1/clustertaint_types.go

```diff
@@ -0,0 +1,141 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:resource:path=clustertaintpolicies,scope="Cluster"
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ClusterTaintPolicy defines how Karmada would taint clusters according
+// to the conditions on the target clusters.
+type ClusterTaintPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec represents the desired behavior of ClusterTaintPolicy.
+	// +required
+	Spec ClusterTaintPolicySpec `json:"spec"`
+}
+
+// ClusterTaintPolicySpec represents the desired behavior of ClusterTaintPolicy.
+type ClusterTaintPolicySpec struct {
+	// TargetClusters specifies the clusters that ClusterTaintPolicy needs
+	// to pay attention to.
+	// For clusters that meet the MatchConditions, Taints will be added.
+	// If targetCluster is not set, any cluster can be selected.
+	// +optional
+	TargetClusters *ClusterAffinity `json:"targetClusters,omitempty"`
+
+	// MatchConditions indicates the conditions to match for triggering
+	// the controller to add taints on the cluster object.
+	// The match conditions are ANDed.
+	// When the MatchConditions no longer match, the taints will be removed.
+	// It can not be empty.
+	// +required
+	MatchConditions []MatchCondition `json:"matchConditions"`
+
+	// Taints specifies the taints that need to be applied to the clusters
+	// which match with TargetClusters.
+	// Distinct ClusterTaintPolicy objects are restricted from operating on
+	// the same taint.
+	// +required
+	Taints []Taint `json:"taints"`
+
+	// AddOnMatchSeconds is the duration in seconds for which the MatchConditions
+	// must be continuously matched before adding the specified taint to the cluster.
+	// This provides stability by ensuring the condition persists long enough
+	// before taking action.
+	// All MatchConditions should remain satisfied with the start time calculated
+	// from each condition's lastTransitionTime.
+	// It requires time synchronization between the controller and the apiserver.
+	// Defaults to 300.
+	// +optional
+	// +kubebuilder:default=300
+	// +kubebuilder:validation:Minimum=1
+	AddOnMatchSeconds *int32 `json:"addOnMatchSeconds,omitempty"`
+
+	// RemoveOnMismatchSeconds is the duration in seconds for which the MatchConditions
+	// must be continuously unmatched before removing the taint from the cluster.
+	// This prevents premature taint removal during brief connection recoveries.
+	// The duration requirement is satisfied when any single condition is no longer met.
+	// The evaluation uses the lastTransitionTime as the start point for duration calculation.
+	// It requires time synchronization between the controller and the apiserver.
+	// Defaults to 180.
+	// +optional
+	// +kubebuilder:default=180
+	// +kubebuilder:validation:Minimum=1
+	RemoveOnMismatchSeconds *int32 `json:"removeOnMismatchSeconds,omitempty"`
```

Comment:
I'd prefer to postpone introducing the `AddOnMatchSeconds` and `RemoveOnMismatchSeconds` in this release, the reason is that they are used to address condition jitter issues, but there is already an anti-jitter mechanism for managing the conditions, so this is not so necessary for this release.

In addition, postponing the two files will significantly reduce the implementation complexity for this version, allowing us to focus on core functionalities.

Reply:
I have written a controller that [considers debouncing](https://github.com/karmada-io/karmada/pull/6368/commits/8a916b588dc57cb14040d175c18237b05d5bbf05#diff-a38d578afb572633578a9760dd8b192d7566329b24514a6c72b579862eacd9bb), and the code complexity is somewhat high. Given the release schedule, should we implement this debouncing feature in version 1.15?

Reply:
+1

Reply:
Updated

Reply:
I didn't find any updated tasks on the [tracking issue](https://github.com/karmada-io/karmada/issues/6317) reflecting this change explicitly, please make sure we don't forget the follow-up work.

Also commenting the to be added&implemented fields instead of removing them would be more helplful for others to understand the whole design, especially regarding these fields are important to the entire cluster taint function.